src/webargs/core.py

```diff
@@ -51,6 +51,12 @@
 DEFAULT_VALIDATION_STATUS: int = 422
 
 
+def _record_arg_name(f: typing.Callable[..., typing.Any], argname: str) -> None:
+    if not hasattr(f, "__webargs_argnames__"):
+        f.__webargs_argnames__ = []  # type: ignore[attr-defined]
+    f.__webargs_argnames__.append(argname)  # type: ignore[attr-defined]
```

Comment:
There's a (admittedly twisted) use case I try to support in flask-smorest when stacking decorators.

Assume a view is decorated once, then used in different places with other decorators (this can be achieved by calling the decorators as normal functions, not the decorator form). Then mutating the list is wrong. It should be replaced instead.

This might not make much sense in the `use_args` case and the likelihood of someone ever doing this is close to zero but it is not forbidden...

Reply:
I had the idea yesterday that if we "just switched to a tuple" this use-case would fall out naturally from that.
When I went to do it today, I found that the only additional tweak which was needed was to put the `__webargs_argnames__` tuple on the `wrapper` built by the decorator, so I think this worked out quite well.

I've added a relevant test-case and it seems pretty good.

I wouldn't necessarily _advertise_ the feature because it's a bit unusual, but it is now supported and tested! :grin: